java/core/src/java/org/apache/orc/impl/ZstdCodec.java

```diff
@@ -214,6 +214,11 @@ public boolean compress(ByteBuffer in, ByteBuffer out,
 
   @Override
   public void decompress(ByteBuffer in, ByteBuffer out) throws IOException {
+    if (in.isDirect() && out.isDirect()) {
```

Comment:
Just a question. When is this used in Apache ORC code base?

Reply:
If we use zerocopy `orc.use.zerocopy=true`, we may use the direct buffer, which can fallback to the original way of reading if the codec doesn't implement `DirectDecompressionCodec`.

https://github.com/apache/orc/blob/4eff23aa3ff6180f00cab472b91c3f67f82d6683/java/core/src/java/org/apache/orc/impl/RecordReaderUtils.java#L600-L607

Reply:
Got it.